fastcrawler/core/spider.py

```diff
@@ -69,46 +99,50 @@ def pending_urls(self) -> set[str]:
         return self._pending_urls
 
     @pending_urls.setter
-    def pending_urls(self, value) -> set[str]:
+    def pending_urls(self, value):
         """Method to overwrite pending urls"""
         self._pending_urls = value
 
+    @property
```

Comment:
you can just use ___len___

Reply:
`get_batch_size` get the batch size, based on the logic. either the self.batch_size, or self._engine_request_limit*2 (double the request limit engine)
so i'm not quite sure how len would be useful in this context. we're not getting len of something. it's just the property to get the user wanted batch size

Reply:
no i'm sating instead of get_batch_size maybe you could've had a dunder len defined in some of your classes

Reply:
Ofc I read. That’s why I told you to look up in comments then, they’ve mentioned why you should return None explicitly

Reply:
get batch size is not a length of something, it’s just to read the configuration value based on the user input, i think you’re getting things mixed up.

Reply:
It’s the index number that should be used somewhere else in code that gets manipulated by the response parser, I couldn’t use len anywhere.

Reply:
I mean 355 liked this article and just 5 people are disagree and that's a good reason?
and by the way the guy who is saying i can think of 100 reasons is a C programmer.

what you are doing here is what is called a dummy return it's usaully bad practice
and in you case the None is not doing anything it's compelety useless

Reply:
Not really, In term of python:
- explicit is better than implicit, which means we should explicitly return None


and here’s some other logics:
- If you decide to change type for function, you can just look up where you returned None
- Easy for debugging, you can breakpoint debugger on return None, and then debug the function
- increasing coding readability

There is no good reason, to not return None.